Library/Homebrew/system_config.rb

```diff
@@ -119,67 +119,40 @@ def dump_verbose_config(f = $stdout)
       else
         f.puts "Core tap: N/A"
       end
-      defaults_hash = {
-        HOMEBREW_PREFIX:        Homebrew::DEFAULT_PREFIX,
-        HOMEBREW_REPOSITORY:    Homebrew::DEFAULT_REPOSITORY,
-        HOMEBREW_CELLAR:        Homebrew::DEFAULT_CELLAR,
-        HOMEBREW_CACHE:         "#{ENV["HOME"]}/Library/Caches/Homebrew",
-        HOMEBREW_LOGS:          "#{ENV["HOME"]}/Library/Logs/Homebrew",
-        HOMEBREW_TEMP:          ENV["HOMEBREW_SYSTEM_TEMP"],
-        HOMEBREW_RUBY_WARNINGS: "-W0",
-      }.freeze
-      boring_keys = %w[
-        HOMEBREW_BROWSER
-        HOMEBREW_EDITOR
-
-        HOMEBREW_ANALYTICS_ID
-        HOMEBREW_ANALYTICS_USER_UUID
-        HOMEBREW_AUTO_UPDATE_CHECKED
-        HOMEBREW_BOTTLE_DEFAULT_DOMAIN
-        HOMEBREW_BOTTLE_DOMAIN
-        HOMEBREW_BREW_FILE
-        HOMEBREW_BREW_GIT_REMOTE
-        HOMEBREW_COMMAND_DEPTH
-        HOMEBREW_CORE_GIT_REMOTE
-        HOMEBREW_CURL
-        HOMEBREW_DISPLAY
-        HOMEBREW_GIT
-        HOMEBREW_GIT_CONFIG_FILE
-        HOMEBREW_LIBRARY
-        HOMEBREW_MACOS_VERSION
-        HOMEBREW_MACOS_VERSION_NUMERIC
-        HOMEBREW_MINIMUM_GIT_VERSION
-        HOMEBREW_RUBY_PATH
-        HOMEBREW_SYSTEM
-        HOMEBREW_SYSTEM_TEMP
-        HOMEBREW_OS_VERSION
-        HOMEBREW_PATH
-        HOMEBREW_PROCESSOR
-        HOMEBREW_PRODUCT
-        HOMEBREW_USER_AGENT
-        HOMEBREW_USER_AGENT_CURL
-        HOMEBREW_VERSION
-      ].freeze
       f.puts "HOMEBREW_PREFIX: #{HOMEBREW_PREFIX}"
-      [:HOMEBREW_CELLAR, :HOMEBREW_CACHE, :HOMEBREW_LOGS, :HOMEBREW_REPOSITORY,
-       :HOMEBREW_TEMP].each do |key|
+      {
+        HOMEBREW_REPOSITORY: Homebrew::DEFAULT_REPOSITORY,
+        HOMEBREW_CELLAR:     Homebrew::DEFAULT_CELLAR,
+        HOMEBREW_CACHE:      "#{ENV["HOME"]}/Library/Caches/Homebrew",
+        HOMEBREW_LOGS:       "#{ENV["HOME"]}/Library/Logs/Homebrew",
+        HOMEBREW_TEMP:       ENV["HOMEBREW_SYSTEM_TEMP"],
```

Comment:
This env no longer exists.

Reply:
good catch, missed that, needs to be a call based off `HOMEBREW_DEFAULT_TEMP`. The above two defaults should be more sensibly handled, too.